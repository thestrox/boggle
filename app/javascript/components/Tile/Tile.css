.tile-container {
    background-color: lightcyan;
    width: 50px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    cursor: pointer;
}

.letter-span {
    font-family: Roboto;
    font-size: 16px;
}
