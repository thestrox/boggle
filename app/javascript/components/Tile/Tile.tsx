import React from "react";
import "./Tile.css"

export type TileProp = {
    letter: string,
    onClick: () => void
}

export default function Tile({ letter, onClick }: TileProp) {

    return (
        <div onClick={onClick} className="tile-container">
            <span className="letter-span">{letter}</span>
        </div>
    );
}