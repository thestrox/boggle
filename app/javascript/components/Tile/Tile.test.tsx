import React from 'react';
import { shallow } from 'enzyme';
import Tile from "./Tile"

function setup() {
    const props = {
        letter: "A",
        onClick: jest.fn()
    };

    const component = shallow(<Tile {...props}/>);
    return {
        props,
        component
    }
}

describe('Tile Component', () => {
    it('should render correctly with given props', () => {
        const {component, props} = setup();
        expect(component.find('div').hasClass('tile-container')).toBe(true);
        let spanNode = component.find('span');
        expect(spanNode.hasClass('letter-span')).toBe(true);
        expect(spanNode.text()).toBe(props.letter);
    });

    it('should call onClick', () => {
        const {component, props} = setup();
        expect(component.find('div').simulate('click'));
        expect(props.onClick).toHaveBeenCalled(); 
    });

    it('should test snapshot renders', () => {
        const {component} = setup();
        expect(component).toMatchSnapshot();
    });
})