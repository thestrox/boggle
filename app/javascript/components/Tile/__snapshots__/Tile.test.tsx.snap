// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Component should test snapshot renders 1`] = `
<div
  className="tile-container"
  onClick={[MockFunction]}
>
  <span
    className="letter-span"
  >
    A
  </span>
</div>
`;
